sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java

```diff
@@ -62,16 +70,100 @@ protected T build(JsonNode node, String path) throws IOException, ConfigurationE
     return super.build(node, path);
   }
 
   private void configure(JsonNode baseNode, MappableField mappableField) {
-    String jsonPathOfField = String.join(".", mappableField.getJsonPathToProperty());
-    addOverride(
-        baseNode,
-        jsonPathOfField,
-        configurationRuntimeContext.getValue(mappableField.getContextKey()));
+    String value = configurationRuntimeContext.getValue(mappableField.getContextKey());
+    try {
+      String jsonPathOfField = String.join(".", mappableField.getJsonPathToProperty());
+      addOverride(baseNode, jsonPathOfField, value);
+    } catch (IllegalArgumentException e) {
+      LOG.info(
+          "{} is not mappable with Dropwizard overrides, trying best guess approach",
+          mappableField,
+          e);
+      addCustomOverride(baseNode, mappableField.getJsonPathToProperty(), value);
+    }
+  }
+
+  private void addCustomOverride(JsonNode rootNode, List<String> path, String value) {
+    try {
+      if (path.isEmpty()) {
+        return;
+      }
+      var propertyName = path.get(0);
+      var valueNow = path.size() == 1;
+
+      // setting values
+      if (valueNow) {
+        addValue(rootNode, value, propertyName);
+        return;
+      }
+
+      // building the hierarchy
+      JsonNode nextNode = addNextNodeInHierarchy(rootNode, path, propertyName);
+      addCustomOverride(nextNode, path.subList(1, path.size()), value);
+
+    } catch (Exception e) {
+      if (e instanceof IllegalArgumentException) {
+        throw e;
+      }
+      LOG.error("Failed to create an override for {}", path, e);
+    }
+  }
+
+  private JsonNode addNextNodeInHierarchy(
+      JsonNode rootNode, List<String> path, String propertyName) {
+    var representsArray = path.size() > 1 && isArrayIndexKey(path.get(1));
+    if (rootNode.isObject()) {
+      return setProperty(rootNode, propertyName, representsArray);
+    } else if (rootNode.isArray() && isArrayIndexKey(propertyName)) {
+      return insertArrayItemIfNotExists((ArrayNode) rootNode, propertyName, representsArray);
+    } else {
+      throw new IllegalArgumentException("Unable to build hierarchy.");
+    }
+  }
+
+  private ContainerNode<? extends ContainerNode<?>> setProperty(
+      JsonNode objectNodeTarget, String propertyName, boolean valueIsArray) {
+    return valueIsArray
+        ? objectNodeTarget.withArrayProperty(propertyName)
+        : objectNodeTarget.withObjectProperty(propertyName);
+  }
+
+  private JsonNode insertArrayItemIfNotExists(
+      ArrayNode targetArrayNode, String nextKey, boolean itemRepresentsArray) {
+    int arrayIndex = Integer.parseInt(nextKey);
+    var nextNode = targetArrayNode.get(arrayIndex);
+    if (nextNode == null || nextNode.isNull() || nextNode.isMissingNode()) {
+      nextNode =
+          itemRepresentsArray
+              ? targetArrayNode.insertArray(arrayIndex)
+              : targetArrayNode.insertObject(arrayIndex);
+    }
+    return nextNode;
+  }
```

Comment:
We do not change values here. So we only have to ensure, that the item exists so that values can be overridden in the next step. The naming `insertArrayItemIfNotExists` is not 100% unambiguous but I did not find a better readable name. Any suggestions?

Reply:
Ah, now I get it. It's just a preparation step.